File: 0643_max_average_subarray_i/MaxAverageSubArrayFixedLength.php
<?php

class MaxAverageSubArrayFixedLength
{

    /**
     * @param int[] $nums
     * @param int $k
     * @return Float
     */
    function findMaxAverage($nums, $k)
    {
        $max_sum = $window_sum = array_sum(array_slice($nums, 0, $k));
        $start = 0;

        for ($i = $k; $i < count($nums); $i++) {
            $window_sum += $nums[$i] - $nums[$i - $k];

            if ($window_sum > $max_sum) {
                $max_sum = $window_sum;
                $start = $i - $k + 1;
            }
        }

        for ($i = $start; $i < $start + $k; $i++) {
            echo "{nums[$i]} = {$nums[$i]} \n";
        }

        return floatval($max_sum / $k);
    }
}

$solution = new MaxAverageSubArrayFixedLength();

$nums = [1, 12, -5, -6, 50, 3];
$k = 4;

print_r($nums);
$answer = $solution->findMaxAverage($nums, $k);

echo "Answer : {$answer}\n";

File: 0643_max_average_subarray_i/MaxAverageSubArrayFixedLength.java

class MaxAverageSubArrayFixedLength {

    public double findMaxAverage(int[] nums, int k) {
        int max_sum;
        int window_sum = 0;
        int start = 0;

        for (int i = 0; i < k; i++) {
            window_sum += nums[i];
        }

        max_sum = window_sum;

        for (int i = k; i < nums.length; i++) {
            window_sum += nums[i] - nums[i - k];

            if (window_sum > max_sum) {
                max_sum = window_sum;
                start = i - k + 1;
            }
        }

        for (int i = start; i < start + k; i++) {
            System.out.println("nums[" + i + "] = " + nums[i]);
        }

        return (double) max_sum / k;
    }

    public static void main(String[] args) {

        MaxAverageSubArrayFixedLength ob = new MaxAverageSubArrayFixedLength();

        int nums[] = {1, 12, -5, -6, 50, 3};
        int k = 4;

        System.out.println("Answer : " + ob.findMaxAverage(nums, k));
    }

}
